source/initial_temperature/lithosphere_mask.cc

```diff
@@ -0,0 +1,195 @@
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    LithosphereMask<dim>::LithosphereMask ()
+      :
+      lab_depths(1, 1.0)
+    {}
+
+
+    template <int dim>
+    void
+    LithosphereMask<dim>::initialize ()
+    {
+    if (LAB_depth_source == File)
+    {
+      const std::string filename = data_directory+LAB_file_name;
+
+      std::cout << "   Loading Ascii data lookup file " << filename << "." << std::endl;
+
+      lab_depths.load_file(filename,this->get_mpi_communicator());
+    }
+    }
+
+    //Read in ascii data - two dimensions so the third column is treated as data
+    template <int dim>
+    double
+    LithosphereMask<dim>::ascii_lab (const Point<2> &position) const
+    {
+      const double lab = lab_depths.get_data(position,0)*1000; //In km
+      return lab;
+    }
+
+    template <int dim>
+    double
+    LithosphereMask<dim>::initial_temperature (const Point<dim> &position) const
+    {
+      double temperature;
+      double depth = this->SimulatorAccess<dim>::get_geometry_model().depth(position);
+
+      if (LAB_depth_source == File)
+        {
+          //Get spherical coordinates for model
+          std::array<double,dim> scoord      = Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
+          const double phi = scoord[1];
+          const double theta = scoord[2];
+          const Point<2> phi_theta (phi, theta);
+
+          //Get lab depth for specific phi and theta
+          const double lab_depth = ascii_lab(phi_theta);
+
+          if (depth <= lab_depth)
+            temperature = lithosphere_temperature;
+          else
+            temperature = std::numeric_limits<double>::quiet_NaN();
+        }
+
+      else if (LAB_depth_source == Value)
+        {
+          if (depth <= max_depth*1000)//in km
+            temperature = lithosphere_temperature;
+          else
+            temperature = std::numeric_limits<double>::quiet_NaN();
+        }
+
+      else
+        {
+          Assert( false, ExcMessage("Invalid method for depth specification method") );
+          return 0.0;
+        }
+
+      return temperature;
+    }
+
+    template <int dim>
+    void
+    LithosphereMask<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.enter_subsection ("Lithosphere Mask");
+        {
+          prm.declare_entry ("Lithosphere temperature", "1600",
+                             Patterns::Double (0),
+                             "The initial temperature within lithosphere, applied above"
+                             "the maximum lithosphere depth.");
+          prm.declare_entry ("Depth specification method", "Value",
+                             Patterns::Selection("File|Value"),
+                             "Method that is used to specify the depth of the lithosphere-asthenosphere boundary.");
+          prm.declare_entry ("Maximum lithosphere depth", "200.0",
```

Comment:
This assumes to be in `km`? Please change that to `m` and document that it is interpreted as meters in the documentation text. We try to stick to SI units as much as possible.